homeassistant/components/telegram_poll.py

```diff
@@ -0,0 +1,160 @@
+CONF_ALLOWED_CHAT_IDS = 'allowed_chat_ids'
+
+REQUIREMENTS = ['python-telegram-bot==5.3.0']
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_ALLOWED_CHAT_IDS): vol.All(cv.ensure_list,
+                                                     [cv.positive_int]),
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup is called when Home Assistant is loading our component."""
+    import telegram
+
+    # instance the Telegram bot
+    bot = telegram.Bot(config[DOMAIN][CONF_API_KEY])
+    websession = async_get_clientsession(hass)
+    allowed_chat_ids = config[DOMAIN][CONF_ALLOWED_CHAT_IDS]
+    update_url = '{0}/getUpdates'.format(bot.base_url)
+    update_id = 0
+    # Flag which is checked while running the incoming loop.
+    running = True
+
+    @callback
+    def _start_bot(_event):
+        """Start the bot."""
+        hass.loop.create_task(check_incoming())
+
+    @callback
+    def _stop_bot(_event):
+        """Stop the bot."""
+        nonlocal running
+        running = False
+
+    hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_START,
+        _start_bot
+    )
+    hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_STOP,
+        _stop_bot
+    )
+
+    @asyncio.coroutine
+    def check_incoming():
+        """"Loop which continuously checks for incoming telegram messages."""
+        while running:
+            try:
+                yield from handle()
+            except AssertionError:
+                _LOGGER.error("incorrect response from telegram server")
+                asyncio.sleep(5)
+
+    @asyncio.coroutine
+    def handle():
+        """" Receiving and processing incoming messages."""
+        nonlocal update_id
+        _updates = yield from get_updates(update_id, 10)
+        for update in _updates['result']:
+            update_id = update['update_id'] + 1
+            try:
+                event, event_data = yield from process_message(update)
+            except UserWarning as warn:
+                _LOGGER.error(warn)
+            else:
+                hass.bus.async_fire(event, event_data)
+
+    @asyncio.coroutine
+    def get_updates(offset, timeout):
+        """Bypass the default getUpdates method to enable asyncio."""
+        response = None
+        data = {'timeout': timeout}
+        if offset:
+            data['offset'] = offset
+        try:
+            response = yield from websession.post(
+                update_url, data=data,
+                headers={'connection': 'keep-alive'}
+            )
+            try:
+                assert response.status == 200
+            except AssertionError:
+                raise HomeAssistantError("Error {0} on {1}".format(
+                    response.status, update_url))
+            try:
+                _json = yield from response.json()
+                return _json
+            except ValueError:
```

Comment:
Since you're not doing anything special, just merge this `try…except` into the already ongoing try block.